angular-workspace/projects/ni/nimble-angular/src/directives/combobox/tests/nimble-combobox-control-value-accessor.directive.spec.ts

```diff
@@ -225,6 +225,32 @@ describe('Nimble combobox control value accessor', () => {
 
             expect(testHostComponent.callbackValue).toEqual(OPTION_NOT_FOUND);
         });
+
+        it('update option value with new model, followed by updating combobox value with same model updates display correctly', async () => {
+            const newModelValue: TestModel = { name: 'newName', value: 100 };
+            testHostComponent.dynamicOption = newModelValue;
+            testHostComponent.selectedOption = newModelValue;
+            fixture.detectChanges();
+            await waitForUpdatesAsync();
+
+            expect(combobox.value).toBe('newName');
+        });
```

Comment:
I'll note that this test would have also passed if I went with an implementation that used `window.requestAnimationFrame()` to process the `list-option` updates (as opposed to the lifecycle hook). I'm not sure there's a way to write the test that would demonstrate that the provided implementation avoids a race condition that we posited would be present with the `window.requestAnimationFrame()` approach.

Reply:
Yea it was something about writing to ngModel before the options were evaluated in a rAF. Nothing immediately came to mind how to exercise it further.